### ISSUE TITLE
Python not found

### ISSUE THREAD

**Issue by jacobacci** [rg3/youtube-dl#16503]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.05.18*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [X I've **verified** and **I assure** that I'm running youtube-dl **2018.05.18**

### Before submitting an *issue* make sure you have:
- [ X] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [X ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [ ] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [X ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---
Please bear with me, if I am not filing this bug report correctly. I am an Ubuntu noob.

I have done a clean install of 18.04 LTS. I then installed youtube-dl using

sudo wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl
sudo chmod a+rx /usr/local/bin/youtube-dl

When I try to use youtube-dl, I get the following error message:

rudolffischer@HP8770w:~$ youtube-dl -U
/usr/bin/env: ‘python’: No such file or directory

Python 3 seems to be installed

rudolffischer@HP8770w:~$ python3
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

I thought youtube-dl is supposed to run with Python3.2+.
Thanks for any help.
Rudi

**Comment by Hrxn** [comment]
What happens if you try `which python`?

(And also `which python3` for completeness)

**Comment by jacobacci** [comment]
Hi,

Thanks for the quick response.

I tried installing Python 2.7 and lo and behold, youtube-dl works.

The installed Python3 Version is listed in my post.

Should I try other Python 3.x versions as well?

**Comment by dstftw** [comment]
Obviously your python configuration is broken. Nothing to do with youtube-dl.

**Comment by jacobacci** [comment]
Thanks for the feedback, although I don’t think the python  configuration is broken after a clean install.

After a bit more reading I suspect that the youtube-dl.exe I installed using wget uses python 2.7 by default. In Ubuntu 18.04 python 2 has been demoted, so it is not present at all any more.  Hence the error message, although python 3 is present.

Two ways to fix this:

-          Install python 2.7 (which is what I did, and it fixed the issue)

-          Install youtube-dl using pip instead of wget to get the version that uses python3

Is this analysis correct? I am trying to learn about Ubuntu, so I am grateful for your help.

Rudi
